letsencrypt/display/ops.py

```diff
@@ -114,23 +114,39 @@ def pick_configurator(
         config, default, plugins, question,
         (interfaces.IAuthenticator, interfaces.IInstaller))
 
-
-def get_email():
+def get_email(more=False, invalid=False):
     """Prompt for valid email address.
 
+    :param bool more: -- explain why the email is strongly advisable, but how to
+        skip it
+    "param bool invalid: -- true if the user just typed something, but it wasn't
+        a valid-looking email
+
     :returns: Email or ``None`` if cancelled by user.
     :rtype: str
 
     """
-    while True:
-        code, email = zope.component.getUtility(interfaces.IDisplay).input(
-            "Enter email address (used for urgent notices and lost key recovery)")
+    msg = "Enter email address (used for urgent notices and lost key recovery)"
+    height = 8
+    if invalid:
+        msg = "There seem to be problems with that address. " + msg
+        height += 1
+    if more:
+        msg += ('\n\nIf you really want to skip this, you can run the client with '
+                '--register-unsafely-without-email but make sure you backup your '
+                'account key from /etc/letsencrypt/accounts.')
+        height += 4
+    code, email = zope.component.getUtility(interfaces.IDisplay).input(msg, height=height)
```

Comment:
This breaks the text interface. We should update both `IDisplay.input` in `letsencrypt/interfaces` and `FileDisplay.input` in `letsencrypt/display/ops.py` to take a height parameter. This is done for other `IDisplay` functions and I see no need to use `kwargs`.